src/core/job.c

```diff
@@ -180,9 +182,15 @@ static void job_merge_into_installed(Job *j, Job *other) {
         assert(j->installed);
         assert(j->unit == other->unit);
 
-        if (j->type != JOB_NOP)
+        if (j->type != JOB_NOP) {
                 assert_se(job_type_merge_and_collapse(&j->type, other->type, j->unit) == 0);
-        else
+
+                /* Keep the oldest ActivationEventInfo, if any */
+                if (j->activation_event_info)
+                        activation_event_info_unref(other->activation_event_info);
```

Comment:
no need to destroy the object, is there? it would be destroyed anyway once the other job is freed..